### ISSUE TITLE
JS Warnings when clicking the Publish button

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#10875]
**Describe the bug**
Not sure this is already tracked. If it's a known issue, of course do feel free to close.

- create a post
- press the Publish button and observe your browser's dev tools console

I'm getting the following warnings:
```
Warning: React does not recognize the `postType` prop on a DOM element. 
Warning: React does not recognize the `hasPublishAction` prop on a DOM element.
Warning: React does not recognize the `isPublished` prop on a DOM element.
Warning: React does not recognize the `isSaving` prop on a DOM element.
Warning: React does not recognize the `isDirty` prop on a DOM element.
```

Seems to me they come from the post publish panel:
https://github.com/WordPress/gutenberg/blob/a8fe865af07812949e5ef97505668e5587c8aefb/packages/editor/src/components/post-publish-panel/index.js#L68

Where `{ ...additionalProps }` are passed to a `<div>` element. Dumping the passed props, they're:
```
aria-label: "Editor publish"
hasPublishAction: [{…}]
isDirty: false
isPublished: false
isSaving: undefined
postType: "post"
role: "region"
tabIndex: -1
```

where I guess only `aria-label`, `role`, and `tabIndex` should be passed because they make this div a navigable region.

**Comment by jhhazelaar** [comment]
Same here. Version 4.0.0

**Comment by desrosj** [comment]
This does not happen for me using WordPress 5.0 with the plugin off. It is happening for me using WordPress 4.9.8 and 5.0 with the plugin activated, though.

**Comment by nosolosw** [comment]
I've noticed that as well and removed most of them in a PR yesterday that will be released with 4.2. https://github.com/WordPress/gutenberg/pull/11093 fixes the one remaining.
